### PR TITLE
[dev-4569] Agency profile v2 subagency table

### PR DESCRIPTION
**Pull request by afnazu**
**High level description:**
A user will be able to see a table of information that includes matches the mock-up
sub-Agency name, award obligations, number transactions, and number awards for each award type

**Technical details:**
Adds subagency table and API integration

**JIRA Ticket:**
[DEV-4569](https://federal-spending-transparency.atlassian.net/browse/DEV-4569)

**Mockup:**
https://bahdigital.invisionapp.com/d/main#/console/14222398/296064992/preview

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [n/a] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [n/a] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
- [n/a] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

**Review by brett-varney — COMMENTED** [review]
Not done yet, but a couple of thoughts so far

File: src/js/apis/agencyV2.js [hunk 1]
```diff
@@ -37,6 +37,11 @@ export const fetchSubagencyCount = (code, fy, params) => apiRequest({
     url: `v2/agency/${code}/sub_agency/count/${fy ? `?fiscal_year=${fy}` : ''}${params ? `&award_type_codes=[${params}]` : ''}`
 });
 
+export const fetchSubagencySpendingList = (code, fy, type, params) => apiRequest({
+    url: `v2/agency/${code}/sub_agency/${fy ? `?fiscal_year=${fy}` : ''}${type ? `&award_type_codes=[${type}]` : ''}`,
```
[the remark this answers is not in the document]
Comment (afnazu) [reply src/js/apis/agencyV2.js:41]: @brett-varney We are currently using awardTypeGroups for table filtering by tab `const typeParam = awardTypeGroups[type];` which passes the associated array based on tab as the type, for example-> contracts: ['A', 'B', 'C', 'D']

File: src/js/components/agencyV2/AgencySection.jsx [hunk 2]
```diff
@@ -13,9 +13,16 @@ const propTypes = {
...
+    <>
+        <h4 className="tooltip__title">Coming Soon</h4>
+        <p className="tooltip__text">The tooltip content for this section is currently under review.</p>
+    </>
+);
+
 const AgencySection = ({
     section,
-    icon = "chart-area",
+    icon,
```
[the remark this answers is not in the document]
Comment (afnazu) [reply src/js/components/agencyV2/AgencySection.jsx]: Thanks good catch - updated to include default so we can avoid any null issues as we are updating the icons!

File: src/js/containers/agencyV2/awardSpending/SubagencyTableContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    const [prevSort, sort, setSort] = useStateWithPrevious('totalObligations');
+    const [prevOrder, order, setOrder] = useStateWithPrevious('desc');
+    const updateSort = (field, direction) => {
+        setSort(field);
+        setOrder(direction);
+    };
+    const [results, setResults] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(false);
+    const request = useRef(null);
```
[the remark this answers is not in the document]
Comment (afnazu) [reply src/js/containers/agencyV2/awardSpending/SubagencyTableContainer.jsx:46]: We are using the ref to set and reset the data being received from the request, would a useRef not be required in this case? open to any suggestions! @brett-varney

**Comment by brett-varney** [comment]
NOTE: there are only 10 agencies in current data, so pagination cannot be checked. OTOH, it also doesn't matter...

**Review by brett-varney — APPROVED** [review]

**Pull request merged**